_state.go

```diff
@@ -1129,6 +1138,8 @@ func (ls *LState) setField(obj LValue, key LValue, value LValue) {
 				ls.RaiseError("attempt to index a non-table object(%v) with key '%s'", curobj.Type().String(), key.String())
 			}
 			ls.RawSet(tb, key, value)
+			// potentially a new key was created, so check the quota
+			tb.CheckQuota(ls)
```

Comment:
This statement should be wrapped by `if` to avoid needless function call that relatively high cost.

```
if ls.tblAllocInfo != nil {
  tb.CheckQuota(ls)
}
```